srml/democracy/src/lib.rs

```diff
@@ -269,18 +291,59 @@ impl<T: Trait> Module<T> {
 
 	/// Get the voters for the current proposal.
 	pub fn tally(ref_index: ReferendumIndex) -> (BalanceOf<T>, BalanceOf<T>, BalanceOf<T>) {
-		Self::voters_for(ref_index).iter()
+		let (approve, against, capital): (BalanceOf<T>, BalanceOf<T>, BalanceOf<T>) = Self::voters_for(ref_index).iter()
 			.map(|voter| (
-				T::Currency::total_balance(voter),
-				Self::vote_of((ref_index, voter.clone())),
+				T::Currency::total_balance(voter), Self::vote_of((ref_index, voter.clone()))
 			))
 			.map(|(bal, vote)|
 				if vote.is_aye() {
 					(bal * BalanceOf::<T>::sa(vote.multiplier() as u64), Zero::zero(), bal)
 				} else {
 					(Zero::zero(), bal * BalanceOf::<T>::sa(vote.multiplier() as u64), bal)
 				}
-			).fold((Zero::zero(), Zero::zero(), Zero::zero()), |(a, b, c), (d, e, f)| (a + d, b + e, c + f))
+			).fold((Zero::zero(), Zero::zero(), Zero::zero()), |(a, b, c), (d, e, f)| (a + d, b + e, c + f));
+		let (del_approve, del_against, del_capital) = Self::tally_delegation(ref_index);
+		(approve + del_approve, against + del_against, capital + del_capital)
+	}
+
+	/// Get the delegated voters for the current proposal.
+	/// I think this goes into a worker once https://github.com/paritytech/substrate/issues/1458 is done. 
+	fn tally_delegation(ref_index: ReferendumIndex) -> (BalanceOf<T>, BalanceOf<T>, BalanceOf<T>) {
+		let mut approve = Zero::zero();
+		let mut against = Zero::zero();
+		let mut capital = Zero::zero();
+		for voter in Self::voters_for(ref_index).iter() {
+			let vote = Self::vote_of((ref_index, voter.clone()));
+			let (votes, balance) = Self::delegated_votes(ref_index, voter.clone(), vote.multiplier());
+			if vote.is_aye() {
+				approve += votes;
+			} else {
+				against += votes;
+			}
+			capital += balance;
+		}
+		(approve, against, capital)
+	}
+
+	fn delegated_votes(ref_index: ReferendumIndex, to: T::AccountId, min_lock_periods: LockPeriods) -> (BalanceOf<T>, BalanceOf<T>) {
+		let mut balance = Zero::zero();
+		let mut votes = Zero::zero();
+		for (delegator, (delegate, periods, r)) in <Delegations<T>>::enumerate() {
+			if ref_index == r && delegate == to {
+				let lock_periods = if min_lock_periods <= periods { min_lock_periods } else { periods };
+				balance += T::Currency::total_balance(&delegator);
+				votes += T::Currency::total_balance(&delegator) * BalanceOf::<T>::sa(lock_periods as u64);
+				let (del_votes, del_balance) = Self::delegated_votes(ref_index, delegator, lock_periods);
+				votes += del_votes;
+				balance += del_balance;
+			}
+		}
+		(votes, balance)
+	}
+
+	/// Indicates if the account is delegating vote.
+	fn is_delegating(who: T::AccountId) -> bool {
```

Comment:
Note that with current approah this might be for some already finished referendum, so not sure if it's entirely valid